amd_openvx_extensions/amd_rpp/source/kernel_rpp.cpp

```diff
@@ -1986,36 +1979,115 @@ vx_node createNode(vx_graph graph, vx_enum kernelEnum, vx_reference params[], vx
     return node;
 }
 
-// looks like following code is not used
-#if 0//ENABLE_OPENCL
-
-vx_status createGraphHandle(vx_node node, RPPCommonHandle **pHandle)
-{
+vx_status createGraphHandle(vx_node node, RPPCommonHandle **pHandle, Rpp32u batchSize, Rpp32u deviceType) {
     RPPCommonHandle *handle = NULL;
     STATUS_ERROR_CHECK(vxGetModuleHandle(node, OPENVX_KHR_RPP, (void **)&handle));
-    if (handle)
-    {
+
+    if (handle) {
         handle->count++;
-    }
-    else
-    {
+    } else {
         handle = new RPPCommonHandle;
         memset(handle, 0, sizeof(*handle));
         handle->count = 1;
-        STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_ATTRIBUTE_AMD_OPENCL_COMMAND_QUEUE, &handle->cmdq, sizeof(handle->cmdq)));
+        
+        if (deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_OPENCL
+            STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_ATTRIBUTE_AMD_OPENCL_COMMAND_QUEUE, &handle->cmdq, sizeof(handle->cmdq)));
+            rppCreateWithStreamAndBatchSize(&data->rppHandle, handle->cmdq, batchSize);
```

Comment:
I thought we are creating rpphandle only once per pipeline

Reply:
Yes, the RPP handle is created once for the first node and for the subsequent nodes if the handle is already created we are just incrementing the pointer, The release is also done in a similar manner